sympy/parsing/sympy_parser.py

```diff
@@ -1256,3 +1273,225 @@ def __getitem__(self, t):
         return tuple([_transformation[_] for _ in i])
 
 T = _T()
+
+
+class ControlledEvaluationException(Exception):
```

Comment:
This should subclass SympifyError.

Reply:
Although maybe not. Is SympifyError used inside of ParseExpr?

Reply:
It's not used in there yet, but I don't see why it couldn't be.